### ISSUE TITLE
Error during app build: Multiple commands produce

### ISSUE THREAD

**Issue by JustJoostNL** [margelo/react-native-filament#251]
I installed react-native-filament according the docs, but I receive this error:

```
Multiple commands produce '/Users/expo/Library/Developer/Xcode/DerivedData/MyApp-ghhktuklewzjozbubcwfbrzgmmoa/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/react_native_filament.framework/Headers/compiler.h'

    duplicate output file '/Users/expo/Library/Developer/Xcode/DerivedData/MyApp-ghhktuklewzjozbubcwfbrzgmmoa/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/react_native_filament.framework/Headers/compiler.h' on task: CpHeader /Users/expo/Library/Developer/Xcode/DerivedData/MyApp-ghhktuklewzjozbubcwfbrzgmmoa/Build/Intermediates.noindex/ArchiveIntermediates/MyApp/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/react_native_filament.framework/Headers/compiler.h /Users/expo/workingdir/build/node_modules/react-native-filament/ios/libs/filament/include/math/compiler.h (in target 'react-native-filament' from project 'Pods')
```

Just FYI, I don't use the new architecture.

Thanks in advance for help!

**Comment by hannojg** [comment]
are you able to provide a reproduction example?

Seems like filament and expo both have a header file called `compiler.h` and those are conflicting 🤔

**Comment by JustJoostNL** [comment]
I will provide a repo with an example soon 🙂

**Comment by JustJoostNL** [comment]
@hannojg https://github.com/JustJoostNL/rn-filament-example

The app fails building with the above errors from earlier.

**Comment by hannojg** [comment]
Do you get the build error on iOS or android?

**Comment by JustJoostNL** [comment]
iOS, haven't tested Android yet.

**Comment by JustJoostNL** [comment]
@hannojg 

I've done some more research, and I don't think the issue is on Expo's side here.

`/Users/expo/workingdir/build/node_modules/react-native-filament/ios/libs/filament/include/camutils/compiler.h`
`/Users/expo/workingdir/build/node_modules/react-native-filament/ios/libs/filament/include/math/compiler.h`

If you look at the paths, you can see that `math` and `camutils` both produce an output file with the same name.
https://github.com/margelo/react-native-filament/blob/main/package/ios/libs/filament/include/camutils/compiler.h
https://github.com/margelo/react-native-filament/blob/main/package/ios/libs/filament/include/math/compiler.h

These are the only libraries that have a `compiler.h` too. So I guess something has to be changed here?

**Comment by antochan** [comment]
same issue here

**Comment by JustJoostNL** [comment]
Is there any update on this? Would love to try out the library!

**Comment by Skipperlla** [comment]
Is there any progress on how it's going to be resolved?

**Comment by hannojg** [comment]
I have't had time yet to look into yet, no

**Comment by stereodenis** [comment]
I think problem is here
<img width="1269" alt="Xcode 2024-10-17 18 17 13" src="https://github.com/user-attachments/assets/7d2dbb90-c379-439d-9fcc-8c4e7a57db3b">

**Comment by Kelier** [comment]
need to fix! 📞

**Comment by Misterr-H** [comment]
hi guys, any update on this?

**Comment by JustJoostNL** [comment]
Is there anyone working on this? I believe this is a pretty important issue because it has impact on many people.

**Comment by JustJoostNL** [comment]
Mmm, that's interesting. In my case, I still get the error. Is someone else able to provide another example project where it is failing?

**Comment by hannojg** [comment]
Did you run the exact same steps as I did in the comment above?

**Comment by JustJoostNL** [comment]
I've used `eas build --platform ios`. Also, maybe this helps: https://github.com/margelo/react-native-filament/issues/251#issuecomment-2255938769?

**Comment by Misterr-H** [comment]
hi @hannojg 
I've created an repo that may helo you reproduce this issue

https://github.com/Misterr-H/rn-filament-bug

do clone
npm i --force
npx expo prebuild -p ios
xed ios

and build

**Comment by CHANDRU6118** [comment]
I am using a bare react native project and i am still getting thir error. Can you help me fix it?

**Comment by CHANDRU6118** [comment]
Yeah the above method tried. but i am still getting error. i have doubt, when i changes made in  node_modules. Is it correct or not.

**Comment by s3rg4sh** [comment]
> Yeah the above method tried. but i am still getting error. i have doubt, when i changes made in node_modules. Is it correct or not. Show some example

Try to rename file node_modules/react_native_filament/ios/libs/filament/include/math/compiler.h
If you will find any paths with this file just change the paths
It's temporary solution

**Comment by JustJoostNL** [comment]
@hannojg Are you able to implement a fix with the above provided reproduction example and the feedback?

**Comment by s3rg4sh** [comment]
You can also check this issue in [repo](https://github.com/s3rg4sh/filattst), this one with bare react native project

**Comment by hannojg** [comment]
Awesome I am able to reproduce with this reproducer:

> @hannojg You can also check this issue in [repo](https://github.com/s3rg4sh/filattst), this one with bare react native project

looking into a fix now 😊

**Comment by hannojg** [comment]
Should be fixed with:

- https://github.com/margelo/react-native-filament/pull/307

please retest! Should be part of version 1.7.0 which should be released in a moment

**Comment by mirceaciu** [comment]
The multiple commands produce error did not reproduce but now I get

```
'react/renderer/consistency/ShadowTreeRevisionConsistencyManager.h' file not found
```

ShadowTreeRevisionConsistencyManager is present in node_modules
